### PR TITLE
Bumping up min OS version to iOS 8 and min XCode version to XCode 7

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

**Comment by bhariharan** [comment]
@khawkins Mind reviewing?

File: build/app_template_files/__NativeSwiftTemplateAppName__/__NativeSwiftTemplateAppName__/Info.plist [hunk 1]
```diff
@@ -15,7 +15,7 @@
         <key>CFBundleIdentifier</key>
         <string>__CompanyIdentifier__.${PRODUCT_NAME:rfc1034identifier}</string>
         <key>CFBundleInfoDictionaryVersion</key>
-        <string>6.0</string>
+        <string>8.0</string>
```
Comment (khawkins) [inline build/app_template_files/__NativeSwiftTemplateAppName__/__NativeSwiftTemplateAppName__/Info.plist]: You don't want to bump this number.  It's actually a separate version number specifying the plist schema version.  6.0 is the latest.

File: build/app_template_files/__NativeTemplateAppName__/__NativeTemplateAppName__/Info.plist [hunk 2]
```diff
@@ -15,7 +15,7 @@
         <key>CFBundleIdentifier</key>
         <string>__CompanyIdentifier__.${PRODUCT_NAME:rfc1034identifier}</string>
         <key>CFBundleInfoDictionaryVersion</key>
-        <string>6.0</string>
+        <string>8.0</string>
```
Comment (khawkins) [inline build/app_template_files/__NativeTemplateAppName__/__NativeTemplateAppName__/Info.plist]: Should stay 6.0.

File: build/app_template_files/__ReactNativeTemplateAppName__/__ReactNativeTemplateAppName__/Info.plist [hunk 3]
```diff
@@ -15,7 +15,7 @@
         <key>CFBundleIdentifier</key>
         <string>__CompanyIdentifier__.${PRODUCT_NAME:rfc1034identifier}</string>
         <key>CFBundleInfoDictionaryVersion</key>
-        <string>6.0</string>
+        <string>8.0</string>
```
Comment (khawkins) [inline build/app_template_files/__ReactNativeTemplateAppName__/__ReactNativeTemplateAppName__/Info.plist]: Should stay 6.0.

File: hybrid/SampleApps/AccountEditor/AccountEditor/AccountEditor-Info.plist [hunk 4]
```diff
@@ -15,7 +15,7 @@
     <key>CFBundleIdentifier</key>
     <string>${BUNDLE_PREFIX}.${PRODUCT_NAME:rfc1034identifier}</string>
     <key>CFBundleInfoDictionaryVersion</key>
-    <string>6.0</string>
+    <string>8.0</string>
```
Comment (khawkins) [inline hybrid/SampleApps/AccountEditor/AccountEditor/AccountEditor-Info.plist]: Should stay 6.0.

File: hybrid/SampleApps/NoteSync/NoteSync/NoteSync-Info.plist [hunk 5]
```diff
@@ -15,7 +15,7 @@
     <key>CFBundleIdentifier</key>
     <string>${BUNDLE_PREFIX}.${PRODUCT_NAME:rfc1034identifier}</string>
     <key>CFBundleInfoDictionaryVersion</key>
-    <string>6.0</string>
+    <string>8.0</string>
```
Comment (khawkins) [inline hybrid/SampleApps/NoteSync/NoteSync/NoteSync-Info.plist]: Should stay 6.0.

File: hybrid/SampleApps/SmartSyncExplorerHybrid/SmartSyncExplorerHybrid/SmartSyncExplorerHybrid-Info.plist [hunk 6]
```diff
@@ -15,7 +15,7 @@
     <key>CFBundleIdentifier</key>
     <string>${BUNDLE_PREFIX}.${PRODUCT_NAME:rfc1034identifier}</string>
     <key>CFBundleInfoDictionaryVersion</key>
-    <string>6.0</string>
+    <string>8.0</string>
```
Comment (khawkins) [inline hybrid/SampleApps/SmartSyncExplorerHybrid/SmartSyncExplorerHybrid/SmartSyncExplorerHybrid-Info.plist]: Should stay 6.0.

File: install.sh [hunk 7]
```diff
@@ -25,7 +25,7 @@ fi
 
 # Check for Xcode minimum version
 XCODE_MIN_VERSION=60
```
Comment (khawkins) [inline install.sh]: This should update to `70` too, yes?

File: libs/SalesforceHybridSDK/HybridPluginTestApp/HybridPluginTestApp-Info.plist [hunk 8]
```diff
@@ -11,7 +11,7 @@
     <key>CFBundleIdentifier</key>
     <string>com.salesforce.mobilesdk.${PRODUCT_NAME:rfc1034identifier}</string>
     <key>CFBundleInfoDictionaryVersion</key>
-    <string>6.0</string>
+    <string>8.0</string>
```
Comment (khawkins) [inline libs/SalesforceHybridSDK/HybridPluginTestApp/HybridPluginTestApp-Info.plist]: Should stay 6.0.

File: libs/SalesforceHybridSDK/HybridPluginTestAppTests/HybridPluginTestAppTests-Info.plist [hunk 9]
```diff
@@ -9,7 +9,7 @@
 	<key>CFBundleIdentifier</key>
 	<string>com.salesforce.mobilesdk.${PRODUCT_NAME:rfc1034identifier}</string>
 	<key>CFBundleInfoDictionaryVersion</key>
-	<string>6.0</string>
+	<string>8.0</string>
```
Comment (khawkins) [inline libs/SalesforceHybridSDK/HybridPluginTestAppTests/HybridPluginTestAppTests-Info.plist]: Should stay 6.0.

File: libs/SalesforceKit/SalesforceKit-iOS/Info.plist [hunk 10]
```diff
@@ -9,7 +9,7 @@
 	<key>CFBundleIdentifier</key>
 	<string>com.salesforce.$(PRODUCT_NAME:rfc1034identifier)</string>
 	<key>CFBundleInfoDictionaryVersion</key>
-	<string>6.0</string>
+	<string>8.0</string>
```
Comment (khawkins) [inline libs/SalesforceKit/SalesforceKit-iOS/Info.plist]: Should stay 6.0.

File: libs/SalesforceKit/SalesforceKit-iOSTests/Info.plist [hunk 11]
```diff
@@ -9,7 +9,7 @@
 	<key>CFBundleIdentifier</key>
 	<string>com.salesforce.$(PRODUCT_NAME:rfc1034identifier)</string>
 	<key>CFBundleInfoDictionaryVersion</key>
-	<string>6.0</string>
+	<string>8.0</string>
```
Comment (khawkins) [inline libs/SalesforceKit/SalesforceKit-iOSTests/Info.plist]: Should stay 6.0.

File: libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetworkiOS/Info.plist [hunk 12]
```diff
@@ -9,7 +9,7 @@
 	<key>CFBundleIdentifier</key>
 	<string>com.salesforce.$(PRODUCT_NAME:rfc1034identifier)</string>
 	<key>CFBundleInfoDictionaryVersion</key>
-	<string>6.0</string>
+	<string>8.0</string>
```
Comment (khawkins) [inline libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetworkiOS/Info.plist]: Should stay 6.0.

File: libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetworkiOSTests/Info.plist [hunk 13]
```diff
@@ -9,7 +9,7 @@
 	<key>CFBundleIdentifier</key>
 	<string>com.salesforce.$(PRODUCT_NAME:rfc1034identifier)</string>
 	<key>CFBundleInfoDictionaryVersion</key>
-	<string>6.0</string>
+	<string>8.0</string>
```
Comment (khawkins) [inline libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetworkiOSTests/Info.plist]: Should stay 6.0.

File: libs/SalesforceOAuth/SalesforceOAuthTest/SalesforceOAuthTest-Info.plist [hunk 14]
```diff
@@ -13,7 +13,7 @@
     <key>CFBundleIdentifier</key>
     <string>com.salesforce.${PRODUCT_NAME:rfc1034identifier}</string>
     <key>CFBundleInfoDictionaryVersion</key>
-    <string>6.0</string>
+    <string>8.0</string>
```
Comment (khawkins) [inline libs/SalesforceOAuth/SalesforceOAuthTest/SalesforceOAuthTest-Info.plist]: Should stay 6.0.

File: libs/SalesforceOAuth/SalesforceOAuthUnitTests/Info.plist [hunk 15]
```diff
@@ -9,7 +9,7 @@
 	<key>CFBundleIdentifier</key>
 	<string>com.salesforce.mobilesdk.$(PRODUCT_NAME:rfc1034identifier)</string>
 	<key>CFBundleInfoDictionaryVersion</key>
-	<string>6.0</string>
+	<string>8.0</string>
```
Comment (khawkins) [inline libs/SalesforceOAuth/SalesforceOAuthUnitTests/Info.plist]: Should stay 6.0.

File: libs/SalesforceRestAPI/SalesforceRestAPITests/Info.plist [hunk 16]
```diff
@@ -9,7 +9,7 @@
 	<key>CFBundleIdentifier</key>
 	<string>com.salesforce.mobilesdk.$(PRODUCT_NAME:rfc1034identifier)</string>
 	<key>CFBundleInfoDictionaryVersion</key>
-	<string>6.0</string>
+	<string>8.0</string>
```
Comment (khawkins) [inline libs/SalesforceRestAPI/SalesforceRestAPITests/Info.plist]: Should stay 6.0.

File: libs/SalesforceRestAPI/SalesforceRestAPIUnitTestApp/SalesforceRestAPIUnitTestApp-Info.plist [hunk 17]
```diff
@@ -11,7 +11,7 @@
     <key>CFBundleIdentifier</key>
     <string>com.salesforce.mobilesdk.${PRODUCT_NAME:rfc1034identifier}</string>
     <key>CFBundleInfoDictionaryVersion</key>
-    <string>6.0</string>
+    <string>8.0</string>
```
Comment (khawkins) [inline libs/SalesforceRestAPI/SalesforceRestAPIUnitTestApp/SalesforceRestAPIUnitTestApp-Info.plist]: Should stay 6.0.

File: libs/SalesforceSDKCommon/SalesforceSDKCommonTests/Info.plist [hunk 18]
```diff
@@ -9,7 +9,7 @@
 	<key>CFBundleIdentifier</key>
 	<string>com.salesforce.mobilesdk.$(PRODUCT_NAME:rfc1034identifier)</string>
 	<key>CFBundleInfoDictionaryVersion</key>
-	<string>6.0</string>
+	<string>8.0</string>
```
Comment (khawkins) [inline libs/SalesforceSDKCommon/SalesforceSDKCommonTests/Info.plist]: Should stay 6.0.

File: libs/SalesforceSDKCore/SalesforceSDKCoreTests/SalesforceSDKCoreTests-Info.plist [hunk 19]
```diff
@@ -9,7 +9,7 @@
 	<key>CFBundleIdentifier</key>
 	<string>com.salesforce.mobilesdk.${PRODUCT_NAME:rfc1034identifier}</string>
 	<key>CFBundleInfoDictionaryVersion</key>
-	<string>6.0</string>
+	<string>8.0</string>
```
Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCoreTests/SalesforceSDKCoreTests-Info.plist]: Should stay 6.0.

File: libs/SalesforceSDKCore/SalesforceSDKCoreUnitTestApp/SalesforceSDKCoreUnitTestApp-Info.plist [hunk 20]
```diff
@@ -11,7 +11,7 @@
     <key>CFBundleIdentifier</key>
     <string>com.salesforce.mobilesdk.${PRODUCT_NAME:rfc1034identifier}</string>
     <key>CFBundleInfoDictionaryVersion</key>
-    <string>6.0</string>
+    <string>8.0</string>
```
Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCoreUnitTestApp/SalesforceSDKCoreUnitTestApp-Info.plist]: Should stay 6.0.

File: libs/SalesforceSecurity/SalesforceSecurityTests/SalesforceSecurityTests-Info.plist [hunk 21]
```diff
@@ -9,7 +9,7 @@
 	<key>CFBundleIdentifier</key>
 	<string>com.salesforce.mobilesdk.${PRODUCT_NAME:rfc1034identifier}</string>
 	<key>CFBundleInfoDictionaryVersion</key>
-	<string>6.0</string>
+	<string>8.0</string>
```
Comment (khawkins) [inline libs/SalesforceSecurity/SalesforceSecurityTests/SalesforceSecurityTests-Info.plist]: Should stay 6.0.

File: libs/SmartSync/SmartSyncTests/SmartSyncTests-Info.plist [hunk 22]
```diff
@@ -9,7 +9,7 @@
     <key>CFBundleIdentifier</key>
     <string>com.salesforce.mobilesdk.${PRODUCT_NAME:rfc1034identifier}</string>
     <key>CFBundleInfoDictionaryVersion</key>
-    <string>6.0</string>
+    <string>8.0</string>
```
Comment (khawkins) [inline libs/SmartSync/SmartSyncTests/SmartSyncTests-Info.plist]: Should stay 6.0.

File: native/SampleApps/RestAPIExplorer/RestAPIExplorer/RestAPIExplorer-Info.plist [hunk 23]
```diff
@@ -13,7 +13,7 @@
     <key>CFBundleIdentifier</key>
     <string>${BUNDLE_PREFIX}.${PRODUCT_NAME:rfc1034identifier}</string>
     <key>CFBundleInfoDictionaryVersion</key>
-    <string>6.0</string>
+    <string>8.0</string>
```
Comment (khawkins) [inline native/SampleApps/RestAPIExplorer/RestAPIExplorer/RestAPIExplorer-Info.plist]: Should stay 6.0.

File: native/SampleApps/SmartSyncExplorer/SmartSyncExplorer/SmartSyncExplorer-Info.plist [hunk 24]
```diff
@@ -11,7 +11,7 @@
     <key>CFBundleIdentifier</key>
     <string>com.salesforce.mobilesdk.${PRODUCT_NAME:rfc1034identifier}</string>
     <key>CFBundleInfoDictionaryVersion</key>
-    <string>6.0</string>
+    <string>8.0</string>
```
Comment (khawkins) [inline native/SampleApps/SmartSyncExplorer/SmartSyncExplorer/SmartSyncExplorer-Info.plist]: Should stay 6.0.

**Comment by khawkins** [comment]
Looks fine over all.  There's probably a bit of follow-on work to update projects, remove pre-iOS 8 code support, etc.  But it looks pretty minimal.

**Comment by bhariharan** [comment]
Thanks for reviewing @khawkins. Things should be lined up now. I've reverted the ```.plist``` version changes I made.

File: hybrid/SampleApps/AccountEditor/AccountEditor/AccountEditor-Info.plist [hunk 25]
```diff
@@ -1,76 +1,73 @@
...
+	<key>CFBundleVersion</key>
+	<string>1.0</string>
+	<key>LSRequiresIPhoneOS</key>
+	<true/>
+	<key>UIRequiredDeviceCapabilities</key>
+	<array>
+		<string>armv7</string>
+	</array>
+	<key>UISupportedInterfaceOrientations</key>
+	<array>
```
Comment (bhariharan) [inline hybrid/SampleApps/AccountEditor/AccountEditor/AccountEditor-Info.plist]: Looks like ```Xcode 7``` changed the formatting on these files. I did it by hand the previous time.

**Comment by khawkins** [comment]
Cool.  Looks like the spacing got goofy in those plist files.  You may want to just revert them, if they didn't change.  Otherwise, looks fine to me.

**Comment by khawkins** [comment]
Oh just saw your comment, well, whatever works. :)

**Pull request merged**